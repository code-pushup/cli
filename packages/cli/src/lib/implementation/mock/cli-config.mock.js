module.exports = {
  persist: { outputPath: 'cli-config-out.json' },
  categories: [],
  plugins: [
    {
      audits: [],
      runner: {
        command: 'bash',
        args: [
          '-c',
          `echo '${JSON.stringify({
            audits: [],
          })}' > cli-config-out.json`,
        ],
        outputPath: 'cli-config-out.json',
      },
      meta: {
        slug: 'execute-plugin',
        name: 'execute plugin',
      },
    },
  ],
};
