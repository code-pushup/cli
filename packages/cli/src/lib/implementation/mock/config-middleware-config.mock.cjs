module.exports = {
  persist: { outputPath: 'tmp/cjs-out.json' },
  plugins: [],
  categories: [],
};
