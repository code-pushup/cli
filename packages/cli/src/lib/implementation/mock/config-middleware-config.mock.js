module.exports = {
  persist: { outputPath: 'tmp/js-out.json' },
  plugins: [],
  categories: [],
};
