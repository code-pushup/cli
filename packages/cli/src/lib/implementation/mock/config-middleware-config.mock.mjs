export default {
  persist: { outputPath: 'tmp/mjs-out.json' },
  plugins: [],
  categories: [],
};
