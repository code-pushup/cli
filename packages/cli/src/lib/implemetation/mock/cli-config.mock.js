module.exports = {
  persist: { outputPath: 'cli-config-out.json' },
  plugins: [
    {
      audits: [],
      runner: {
        command: 'bash',
        args: [
          '-c',
          `echo '${JSON.stringify({
            audits: [],
          })}' > cli-config-out.json`,
        ],
        outputPath: 'cli-config-out.json',
      },
      groups: [],
      meta: {
        slug: 'execute-plugin',
        name: 'execute plugin',
        type: 'static-analysis',
      },
    },
  ],
  categories: [],
};
