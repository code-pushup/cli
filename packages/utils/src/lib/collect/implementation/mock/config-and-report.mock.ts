import { PluginConfig } from '@quality-metrics/models';
import {
  mockCategory,
  mockConfig,
  mockPluginConfig,
  mockReport,
} from './schema-helper.mock';

export const nxValidatorsPlugin: () => PluginConfig = () => ({
  runner: {
    command: 'bun',
    args: ['--help'],
    outputPath: 'nx-validators-plugin-runner-output.json',
  },
  meta: {
    // package and version optional
    slug: 'nx-validators',
    name: 'NxValidatorsPlugin',
    docsUrl: `https://github.com/BioPhoton/nx-validators/tree/main/README.md`,
  },
  groups: [
    {
      slug: 'use-nx-tooling',
      description: 'Nx Tooling compliance checks',
      title: 'Use Nx Tooling',
      refs: [
        {
          slug: 'check-version-mismatch',
          weight: 0,
        },
        {
          slug: 'use-nx-cloud',
          weight: 0,
        },
        {
          slug: 'check-gulp-usage',
          weight: 0,
        },
      ],
    },
    {
      slug: 'use-quality-tooling',
      description: 'Eslint, tsconfig, Prettier, ... Tooling compliance checks',
      title: 'Use Quality Tooling',
      refs: [
        {
          slug: 'check-eslint-config',
          weight: 0,
        },
        {
          slug: 'check-prettier-config',
          weight: 0,
        },
      ],
    },
    {
      slug: 'normalize-typescript-config',
      description:
        'Ensure that all typescript configurations are following the Nx standards',
      title: 'Normalize Typescript configuration',
      refs: [
        {
          slug: 'check-root-tsconfig-base',
          weight: 0,
        },
        {
          slug: 'check-tsconfig-paths',
          weight: 0,
        },
      ],
    },
    {
      slug: 'use-workspace-layout',
      description:
        'Folder structure, project.json files, ... validate the workspace layout',
      title: 'Use Workspace Layout',
      refs: [
        {
          slug: 'use-project-json',
          weight: 0,
        },
      ],
    },
  ],
  audits: [
    {
      slug: 'check-prettier-config',
      title: 'Check Prettier Configuration',
      label: 'Check Prettier Configuration',
      description:
        'The goal of that validator is to check that prettier is correctly configured in the current repository.',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-prettier-config/Readme.md',
    },
    {
      slug: 'check-tslint-not-used',
      title: 'Check TSLint is not Used',
      label: 'Check TSLint is not Used',
      description: '`TBD`',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-tslint-not-used/Readme.md',
    },
    {
      slug: 'use-single-entry-file',
      title: 'Use single entry file',
      label: 'Use single entry file',
      description: '`TBD`',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/use-single-entry-file/Readme.md',
    },
    {
      slug: 'check-version-mismatch',
      title: 'Check Version Mismatch',
      label: 'Check Version Mismatch',
      description:
        'This validator checks that all the packages that are used in the repository are aligned with the dependency from the monorepo dependencies.',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-version-mismatch/Readme.md',
    },
    {
      slug: 'use-project-json',
      title: 'Use project.json Files',
      label: 'Use project.json Files',
      description:
        'This validator checks that everything that could be considered as a Nx project should contain a file `project.json`.',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/use-project-json/Readme.md',
    },
    {
      slug: 'use-nx-cloud',
      title: 'Use Nx Cloud',
      label: 'Use Nx Cloud',
      description:
        'This validator checks that the workspace is using Nx Cloud.',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/use-nx-cloud/Readme.md',
    },
    {
      slug: 'use-workspace-folder-structure',
      title: 'Use workspace folder structure',
      label: 'Use workspace folder structure',
      description: '`TBD`',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/use-workspace-folder-structure/Readme.md',
    },
    {
      slug: 'check-tsconfig-paths',
      title: 'Check tsconfig paths',
      label: 'Check tsconfig paths',
      description:
        'This validator checks that the main typescript configuration is containing valid path configurations. To do so, wildcard paths are not allowed.',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-tsconfig-paths/Readme.md',
    },
    {
      slug: 'check-root-tsconfig-base',
      title: 'Check root tsconfig.base.json',
      label: 'Check root tsconfig.base.json',
      description:
        'The goal of that validator is to check that the main typescript configurations is correctly configured in the current repository.',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-root-tsconfig-base/Readme.md',
    },
    {
      slug: 'check-tsconfig-per-project',
      title: 'Check Boundaries Linting Configuration',
      label: 'Check Boundaries Linting Configuration',
      description: '`TBD`',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-tsconfig-per-project/Readme.md',
    },
    {
      slug: 'check-ts-compiler-options',
      title: 'Check Typescript Compiler Options',
      label: 'Check Typescript Compiler Options',
      description: '`TBD`',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-ts-compiler-options/Readme.md',
    },
    {
      slug: 'use-publishable-library',
      title: 'Use publishable library',
      label: 'Use publishable library',
      description: '`TBD`',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/use-publishable-library/Readme.md',
    },
    {
      slug: 'check-external-imports',
      title: 'Check External Imports',
      label: 'Check External Imports',
      description: '`TBD`',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-external-imports/Readme.md',
    },
    {
      slug: 'check-gulp-usage',
      title: 'Check Gulp Usage',
      label: 'Check Gulp Usage',
      description:
        'In a Nx architecture, Gulp is not needed anymore because [Nx Tasks](https://nx.dev/core-features/run-tasks) will be used instead to standardize the executions you can have on each project.',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-gulp-usage/Readme.md',
    },
    {
      slug: 'check-eslint-config',
      title: 'Check ESLint Configs',
      label: 'Check ESLint Configs',
      description:
        "This validator checks for the `eslint` package existence in the target's project repository. It also compares eslint configuration with the expected one, and generates a report with the diff object of the configurations.",
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-eslint-config/Readme.md',
    },
    {
      slug: 'check-boundaries-config',
      title: 'Check Boundaries Linting Configuration',
      label: 'Check Boundaries Linting Configuration',
      description: '`TBD`',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-boundaries-config/Readme.md',
    },
    {
      slug: 'check-webpack-config',
      title: 'Check Webpack Config',
      label: 'Check Webpack Config',
      description: '',
      docsUrl:
        'https://github.com/BioPhoton/nx-validators/tree/main/lib/workspace-validation/validators/check-webpack-config/Readme.md',
    },
  ],
});

export const nxValidatorsOnlyConfig = mockConfig();
nxValidatorsOnlyConfig.plugins = [nxValidatorsPlugin()];
nxValidatorsOnlyConfig.categories = [
  mockCategory({
    categorySlug: 'nx-quality',
    pluginSlug: 'nx-validators',
    auditSlug: nxValidatorsPlugin().audits.map(({ slug }) => slug),
  }),
];

export const nxValidatorsOnlyReport = mockReport();

// ==============

const pluginSlug = ['plg-0', 'plg-1', 'plg-2'];
const auditSlug0 = ['0a', '0b', '0c', '0d'];
const auditSlug1 = ['1a', '1b', '1c'];
const auditSlug2 = ['2a', '2b', '2c', '2d', '2e'];

export const dummyConfig = mockConfig();
dummyConfig.plugins = [
  mockPluginConfig({ pluginSlug: pluginSlug[0], auditSlug: auditSlug0 }),
  mockPluginConfig({ pluginSlug: pluginSlug[1], auditSlug: auditSlug1 }),
  mockPluginConfig({ pluginSlug: pluginSlug[2], auditSlug: auditSlug2 }),
];

dummyConfig.categories = [
  mockCategory({
    pluginSlug: pluginSlug[0],
    categorySlug: 'performance',
    auditSlug: auditSlug0,
  }),
  mockCategory({
    pluginSlug: pluginSlug[1],
    categorySlug: 'a11y',
    auditSlug: auditSlug1,
  }),
  mockCategory({
    pluginSlug: pluginSlug[2],
    categorySlug: 'seo',
    auditSlug: auditSlug2,
  }),
];

export const dummyReport = mockReport({
  pluginSlug: pluginSlug[0],
  auditSlug: auditSlug0,
});
