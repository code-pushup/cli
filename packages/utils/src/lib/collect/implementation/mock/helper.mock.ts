import { vi } from 'vitest';

const _log = console.log;
let mocked = false;
export function mockConsole(
  log: (...args: string[]) => void = (..._: string[]) => {
    void 0;
  },
) {
  console.log = vi.fn(log);
  mocked = true;
}

export function unmockConsole() {
  console.log = _log;
  mocked = false;
}
