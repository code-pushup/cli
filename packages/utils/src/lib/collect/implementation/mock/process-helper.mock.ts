import { vi } from 'vitest';
import { ProcessConfig } from '../execute-process';
import { join } from 'path';

const asyncProcessPath = join(__dirname, './execute-process.mock.mjs');

/**
 * Helps to get a async process runner config for testing.
 *
 * @param cfg
 */
export function getAsyncProcessRunnerConfig(
  cfg: Partial<ProcessConfig> & {
    throwError?: boolean;
    interval?: number;
    runs?: number;
    outputPath?: string;
  } = { throwError: false },
) {
  const outputPath = cfg?.outputPath || './tmp/out-async-runner.json';
  const args = [
    asyncProcessPath,
    cfg?.interval ? cfg.interval + '' : '10',
    cfg?.runs ? cfg.runs + '' : '4',
    cfg?.throwError ? '1' : '0',
    outputPath,
  ];
  return { command: 'node', args, outputPath };
}

/**
 * Helps to get a sync process runner config for testing.
 *
 * @param cfg
 */
export function getSyncProcessRunnerConfig(
  cfg: Partial<ProcessConfig> & {
    throwError?: boolean;
    outputPath?: string;
  } = { throwError: false },
) {
  return {
    command: 'bash',
    args: [
      '-c',
      `echo '${JSON.stringify({
        audits: cfg.throwError
          ? ({ throwError: cfg.throwError } as unknown)
          : [],
      })}' > ${cfg.outputPath}`,
    ],
    outputPath: cfg.outputPath,
  };
}

export function mockProcessConfig(
  processConfig: Partial<ProcessConfig>,
): ProcessConfig {
  return {
    ...{ command: 'dummy-string', args: [], outputPath: './out.json' },
    ...processConfig,
    observer: spyObserver(),
  };
}

/**
 * Helps to set up spy observers for testing.
 */
export function spyObserver() {
  const nextSpy = vi.fn();
  const errorSpy = vi.fn();
  const completeSpy = vi.fn();
  return {
    next: nextSpy,
    error: errorSpy,
    complete: completeSpy,
  };
}
