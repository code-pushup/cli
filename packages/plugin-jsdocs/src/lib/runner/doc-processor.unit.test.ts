import type { ClassDeclaration, VariableStatement } from 'ts-morph';
import { nodeMock } from '../../../mocks/node.mock.js';
import { sourceFileMock } from '../../../mocks/source-files.mock.js';
import {
  getAllNodesFromASourceFile,
  getClassNodes,
  getDocumentationReport,
  getVariablesInformation,
  mergeDocumentationReports,
} from './doc-processor.js';
import type { DocumentationReport } from './models.js';

describe('getDocumentationReport', () => {
  it('should produce a full report', () => {
    const results = getDocumentationReport([
      sourceFileMock('test.ts', {
        functions: { 1: true, 2: true, 3: true },
        classes: { 4: false, 5: false, 6: true },
        enums: { 7: true, 8: false, 9: false },
        types: { 10: false, 11: false, 12: true, 40: true },
        interfaces: { 13: true, 14: true, 15: false },
        properties: { 16: false, 17: false, 18: false },
        variables: { 22: true, 23: true, 24: true },
      }),
    ]);
    expect(results).toMatchSnapshot();
  });

  it('should accept array of source files', () => {
    const results = getDocumentationReport([
      sourceFileMock('test.ts', { functions: { 1: true, 2: true, 3: false } }),
    ]);
    expect(results).toBeDefined();
  });

  it('should count nodes correctly', () => {
    const results = getDocumentationReport([
      sourceFileMock('test.ts', { functions: { 1: true, 2: true, 3: false } }),
    ]);

    expect(results.functions.nodesCount).toBe(3);
  });

  it('should collect uncommented nodes issues', () => {
    const results = getDocumentationReport([
      sourceFileMock('test.ts', { functions: { 1: true, 2: false, 3: false } }),
    ]);

    expect(results.functions.issues).toHaveLength(2);
  });

  it('should collect valid issues', () => {
    const results = getDocumentationReport([
      sourceFileMock('test.ts', { functions: { 1: false } }),
    ]);

    expect(results.functions.issues).toStrictEqual([
      {
        line: 1,
        file: 'test.ts',
        type: 'functions',
        name: 'test',
      },
    ]);
  });

  it('should calculate coverage correctly', () => {
    const results = getDocumentationReport([
      sourceFileMock('test.ts', { functions: { 1: true, 2: false } }),
    ]);

    expect(results.functions.coverage).toBe(50);
  });
});

describe('mergeDocumentationReports', () => {
  const emptyResult: DocumentationReport = {
    enums: { nodesCount: 0, issues: [] },
    interfaces: { nodesCount: 0, issues: [] },
    types: { nodesCount: 0, issues: [] },
    functions: { nodesCount: 0, issues: [] },
    variables: { nodesCount: 0, issues: [] },
    classes: { nodesCount: 0, issues: [] },
    methods: { nodesCount: 0, issues: [] },
    properties: { nodesCount: 0, issues: [] },
  };

  it.each([
    'enums',
    'interfaces',
    'types',
    'functions',
    'variables',
    'classes',
    'methods',
    'properties',
  ])('should merge results on top-level property: %s', type => {
    const secondResult = {
      [type]: {
        nodesCount: 1,
        issues: [{ file: 'test2.ts', line: 1, name: 'test2', type }],
      },
    };

    const results = mergeDocumentationReports(
      emptyResult,
      secondResult as Partial<DocumentationReport>,
    );
    expect(results).toStrictEqual(
      expect.objectContaining({
        [type]: {
          nodesCount: 1,
          issues: [{ file: 'test2.ts', line: 1, name: 'test2', type }],
        },
      }),
    );
  });

  it('should merge empty results', () => {
    const results = mergeDocumentationReports(emptyResult, emptyResult);
    expect(results).toStrictEqual(emptyResult);
  });

  it('should merge second level property nodesCount', () => {
    const results = mergeDocumentationReports(
      {
        ...emptyResult,
        enums: { nodesCount: 1, issues: [] },
      },
      {
        enums: { nodesCount: 1, issues: [] },
      },
    );
    expect(results.enums.nodesCount).toBe(2);
  });

  it('should merge second level property issues', () => {
    const results = mergeDocumentationReports(
      {
        ...emptyResult,
        enums: {
          nodesCount: 0,
          issues: [
            {
              file: 'file.enum-first.ts',
              line: 6,
              name: 'file.enum-first',
              type: 'enums',
            },
          ],
        },
      },
      {
        enums: {
          nodesCount: 0,
          issues: [
            {
              file: 'file.enum-second.ts',
              line: 5,
              name: 'file.enum-second',
              type: 'enums',
            },
          ],
        },
      },
    );
    expect(results.enums.issues).toStrictEqual([
      {
        file: 'file.enum-first.ts',
        line: 6,
        name: 'file.enum-first',
        type: 'enums',
      },
      {
        file: 'file.enum-second.ts',
        line: 5,
        name: 'file.enum-second',
        type: 'enums',
      },
    ]);
  });
});

describe('getClassNodes', () => {
  it('should return all nodes from a class', () => {
    const nodeMock1 = nodeMock({
      coverageType: 'classes',
      line: 1,
      file: 'test.ts',
      isCommented: false,
    });

    const classNodeSpy = vi.spyOn(nodeMock1, 'getMethods');
    const propertyNodeSpy = vi.spyOn(nodeMock1, 'getProperties');

    getClassNodes([nodeMock1] as unknown as ClassDeclaration[]);

    expect(classNodeSpy).toHaveBeenCalledTimes(1);
    expect(propertyNodeSpy).toHaveBeenCalledTimes(1);
  });
});

describe('getVariablesInformation', () => {
  it('should process variable statements correctly', () => {
    const mockDeclaration = {
      getName: () => 'testVariable',
    };

    const mockVariableStatement = {
      getKind: () => 'const',
      getJsDocs: () => ['some docs'],
      getStartLineNumber: () => 42,
      getDeclarations: () => [mockDeclaration],
    };

    const result = getVariablesInformation([
      mockVariableStatement as unknown as VariableStatement,
    ]);

    expect(result).toHaveLength(1);
    expect(result[0]).toEqual({
      getKind: expect.any(Function),
      getJsDocs: expect.any(Function),
      getStartLineNumber: expect.any(Function),
      getName: expect.any(Function),
    });
    // It must be defined
    expect(result[0]!.getName()).toBe('testVariable');
    expect(result[0]!.getKind()).toBe('const');
    expect(result[0]!.getJsDocs()).toEqual(['some docs']);
    expect(result[0]!.getStartLineNumber()).toBe(42);
  });

  it('should handle multiple declarations in a single variable statement', () => {
    const mockDeclarations = [
      { getName: () => 'var1' },
      { getName: () => 'var2' },
    ];

    const mockVariableStatement = {
      getKind: () => 'let',
      getJsDocs: () => [],
      getStartLineNumber: () => 10,
      getDeclarations: () => mockDeclarations,
    };

    const result = getVariablesInformation([
      mockVariableStatement as unknown as VariableStatement,
    ]);

    expect(result).toHaveLength(2);
    // They must be defined
    expect(result[0]!.getName()).toBe('var1');
    expect(result[1]!.getName()).toBe('var2');
    expect(result[0]!.getKind()).toBe('let');
    expect(result[1]!.getKind()).toBe('let');
  });

  it('should handle empty variable statements array', () => {
    const result = getVariablesInformation([]);
    expect(result).toHaveLength(0);
  });

  it('should handle variable statements without declarations', () => {
    const mockVariableStatement = {
      getKind: () => 'const',
      getJsDocs: () => [],
      getStartLineNumber: () => 1,
      getDeclarations: () => [],
    };

    const result = getVariablesInformation([
      mockVariableStatement as unknown as VariableStatement,
    ]);
    expect(result).toHaveLength(0);
  });
});

describe('getAllNodesFromASourceFile', () => {
  it('should combine all node types from a source file', () => {
    const mockSourceFile = sourceFileMock('test.ts', {
      functions: { 1: true },
      classes: { 2: false },
      types: { 3: true },
      enums: { 4: false },
      interfaces: { 5: true },
    });

    const result = getAllNodesFromASourceFile(mockSourceFile);

    expect(result).toHaveLength(5);
  });

  it('should handle empty source file', () => {
    const mockSourceFile = sourceFileMock('empty.ts', {});

    const result = getAllNodesFromASourceFile(mockSourceFile);

    expect(result).toHaveLength(0);
  });

  it('should handle source file with only functions', () => {
    const mockSourceFile = sourceFileMock('functions.ts', {
      functions: { 1: true, 2: false, 3: true },
    });

    const result = getAllNodesFromASourceFile(mockSourceFile);

    expect(result).toHaveLength(3);
  });
});
