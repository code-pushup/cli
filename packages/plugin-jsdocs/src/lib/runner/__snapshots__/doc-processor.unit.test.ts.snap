// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getDocumentationReport > should produce a full report 1`] = `
{
  "classes": {
    "coverage": 33.33,
    "issues": [
      {
        "file": "test.ts",
        "line": 4,
        "name": "test",
        "type": "classes",
      },
      {
        "file": "test.ts",
        "line": 5,
        "name": "test",
        "type": "classes",
      },
    ],
    "nodesCount": 3,
  },
  "enums": {
    "coverage": 33.33,
    "issues": [
      {
        "file": "test.ts",
        "line": 8,
        "name": "test",
        "type": "enums",
      },
      {
        "file": "test.ts",
        "line": 9,
        "name": "test",
        "type": "enums",
      },
    ],
    "nodesCount": 3,
  },
  "functions": {
    "coverage": 100,
    "issues": [],
    "nodesCount": 3,
  },
  "interfaces": {
    "coverage": 66.67,
    "issues": [
      {
        "file": "test.ts",
        "line": 15,
        "name": "test",
        "type": "interfaces",
      },
    ],
    "nodesCount": 3,
  },
  "methods": {
    "coverage": 100,
    "issues": [],
    "nodesCount": 0,
  },
  "properties": {
    "coverage": 100,
    "issues": [],
    "nodesCount": 0,
  },
  "types": {
    "coverage": 50,
    "issues": [
      {
        "file": "test.ts",
        "line": 10,
        "name": "test",
        "type": "types",
      },
      {
        "file": "test.ts",
        "line": 11,
        "name": "test",
        "type": "types",
      },
    ],
    "nodesCount": 4,
  },
  "variables": {
    "coverage": 100,
    "issues": [],
    "nodesCount": 0,
  },
}
`;
